pyrealm/splash/splash.py

```diff
@@ -72,6 +72,9 @@ def __init__(
         self.shape: tuple = check_input_shapes(elv, lat, sf, tc, pn)
         """The array shape of the input variables"""
 
+        if len(dates) != self.shape[0]:
+            raise ValueError("Number of dates must match the first dimension of inputs")
+
```

Comment:
Good catch - can you add units tests to demonstrate though.

Reply:
Yes I will add unit tests for this check.

Reply:
I've added a check in the `splash_model` fixture in the unit test